crates/common/src/lib.rs

```diff
@@ -534,6 +534,15 @@ pub fn calculate_class_commitment_leaf_hash(
     )
 }
 
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq)]
+pub struct ReorgCounter(i64);
+
+impl ReorgCounter {
+    pub fn new(value: i64) -> Self {
+        Self(value)
+    }
+}
+
```

Comment:
This could actually move entirely to `storage` I think?

Reply:
Done.